ChegeBryan/politico
# #165906293 Fetch-Api User signup

#### What does this PR do?

Enable users to signup using the form UI

#### Description of Task to be completed?

- [x] Add client-side input validations.
- [x] Add client-side server communication for user signup (fetch-api) 

#### How should this be manually tested?
To test locally clone the repository to your local machine - check
[README.md](README.md) for instructions.  Check out to this branch `ft-fetchapi-user-signup-165906293`. Navigate to the index page from your local server click on signup link and fill in the required details in the form and submit. If the signup was successful you will be redirected to the user dashboard.

#### What are the relevant pivotal tracker stories?
[#165906293](https://www.pivotaltracker.com/story/show/165906293)
